### PR TITLE
Use EnsureObjectV2

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Use EnsureObjectV2

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/controller/monitoring/monitoring_controller.go [hunk 1]
```diff
@@ -49,7 +48,6 @@ type Controller struct {
 	kubeClient              kubernetes.Interface
 	dynamicClient           ctrlruntimeclient.Client
 	userClusterConnProvider userClusterConnectionProvider
-	dynamicCache            ctrlruntimecache.Cache
```
Comment (mrIncompetent) [inline api/pkg/controller/monitoring/monitoring_controller.go:51]: We might miss some caches - but that's solves after i migrated this controller to the controller runtime

File: api/pkg/controller/usercluster/reconciler.go [hunk 2]
```diff
@@ -148,7 +148,7 @@ func (r *reconciler) reconcileRoles(ctx context.Context) error {
 		machinecontroller.KubePublicRoleCreator(),
 	}
 
-	if err := reconciling.ReconcileRoles(creators, metav1.NamespacePublic, r.Client, r.cache); err != nil {
+	if err := reconciling.ReconcileRoles(ctx, creators, metav1.NamespacePublic, r); err != nil {
```
Comment (alvaroaleman) [inline api/pkg/controller/usercluster/reconciler.go]: That does work, but for clarity, can we pass `r.Client` instead?

**Comment by alvaroaleman** [comment]
The build of `Image-loader` fails :(

**Comment by mrIncompetent** [comment]
@alvaroaleman i had to rebase due to conflicts, PTAL

**Pull request merged**